### ISSUE TITLE
Compiling Grapical

### ISSUE THREAD

**Issue by trevorKirkby** [Chris-plus-alphanumericgibberish/dnethack#22]
It looks like the graphical version of this is a little bit out of date. Dnethack is actively in development and I guess the graphics part just fell behind, because it probably shouldn't be priority.

But the fact remains that the monsters.txt, objects.txt, and other.txt files are missing a lot of tiles that are coded into the tty version.

So I've attached a new set of monsters.txt, objects.txt, and other.txt that should be at least somewhat functional. I didn't actually create a single new tile for these, which means that new items and monsters either have the same graphics as existing ones, or that I grabbed a graphic from slash'em. But in my opinion its better than nothing.

Using those txt files I successfully compiled a graphical dnethackw.exe, although (likely due to the fact that I'm bad at compiling C on windows) the exe isn't actually working.

But with those .txt files it should definitely be possible to make a working, up-to-date tile version. Please consider making such a version in the repo with the files I've attached.

[monsters.txt](https://github.com/Chris-plus-alphanumericgibberish/dnethack/files/1136895/monsters.txt)
[objects.txt](https://github.com/Chris-plus-alphanumericgibberish/dnethack/files/1136894/objects.txt)
[other.txt](https://github.com/Chris-plus-alphanumericgibberish/dnethack/files/1136896/other.txt)

**Comment by Ozma777** [comment]
as far as I know trying to compile in MinGW on a win 32 machine still gives this error too:

C:\GAMES\temp\dnethack\src>mingw32-make -f makefile.gcc install 1>out.txt
../win/win32/mswproc.c: In function 'mswin_player_selection':
../win/win32/mswproc.c:269:5: error: too few arguments to function 'randrole'
../include/extern.h:1981:7: note: declared here
../win/win32/mswproc.c: In function 'prompt_for_player_selection':
../win/win32/mswproc.c:381:7: error: too few arguments to function 'randrole'
../include/extern.h:1981:7: note: declared here
../win/win32/mswproc.c:417:7: error: too few arguments to function 'randrole'
../include/extern.h:1981:7: note: declared here
mingw32-make: *** [o/mswproc.o] Error 1

REPLACING with dNAO code does NOT fix this (although it did fix tty here https://github.com/Chris-plus-alphanumericgibberish/dnethack/issues/16), maybe standard nethack copy/paste function?
I'll try that and see, if it works I'll try and email Chris-plus the NEW dnethack-master.zip with precompiled dnethack.zip and dnethackW.zip AND updated working source.

edit: no nethack code replacement in that function does NOT fix it.

**Comment by trevorKirkby** [comment]
Yes, I encountered that error. The debug output says what the problem is. The randrole() function expects an argument and doesn't get one. Since I never use random roles anyway, I took the lazy route and just destroyed  the randrole function and replaced it with something that always returns a single value.

However that's hardly a good long-term fix. It's just a hack to make it compile. The only reason I did it that way is because I didn't want to muck around too much with code I didn't understand too well.

**Comment by Ozma777** [comment]
do you think that the dNAO code might help? The last version (before Yuki-Onna and Bat races) I fixed the code so it would compile by changing the broken parts (that errored out) to the dNAO versions and then DrowNoble.des (and the others) to Drow Noble.des (what it was looking for in the dat folder).

This new source precompiled for windows here:
https://www.sendspace.com/file/dxxlw9

Worked without me having to change anything so maybe the graphical works too...
Although the zip files dNethack.zip and dNethackW.zip HERE are missing ALL the new roles, races, and fixes.
